src/main/java/org/candlepin/model/ContentAccessPayloadCurator.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2009 - 2025 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.model;
+
+import javax.inject.Singleton;
+import javax.persistence.NoResultException;
+
+/**
+ * Curator responsible for datastore operation for the {@link ContentAccessPayload} entity.
+ */
+@Singleton
+public class ContentAccessPayloadCurator extends AbstractHibernateCurator<ContentAccessPayload> {
+
+    public ContentAccessPayloadCurator() {
+        super(ContentAccessPayload.class);
+    }
+
+    /**
+     * Retrieves a {@link ContentAccessPayload} based on a provided {@link Owner} ID and payload key.
+     *
+     * @param ownerId
+     *  the ID of the owner which owns the content access payload
+     *
+     * @param payloadKey
+     *  the payload of key of the content access payload
+     *
+     * @return the content access payload that has the provided payload key and belongs to the provided owner,
+     *  or null if the payload cannot be found
+     */
+    public ContentAccessPayload getContentAccessPayload(String ownerId, String payloadKey) {
+        if (ownerId == null || ownerId.isBlank()) {
```

Comment:
As a note: these validations aren't bad, but in this particular case it isn't necessary. If either field is null or blank, the resultant query will look like `... WHERE owner_id = NULL AND payload_key = ''`; both of which will result in a valid but effectively dead query that won't return anything. In either case we end up on the null path.

This isn't a request to remove them, just an explanation